src/smpi/smpi_replay.cpp

```diff
@@ -998,44 +1223,76 @@ void smpi_replay_run(int *argc, char***argv){
     XBT_DEBUG("Force context switch by smpi_execute_flops  - Sleeping for 0.0 flops ");
     smpi_execute_flops(0.0);
   }
+}
 
-  /* Actually run the replay */
-  xbt_replay_action_runner(*argc, *argv);
 
+/*
+ * This function only initializes smpi_replay. It does not start the replay.
+ * You should call this one if you want to be able to register new actions
+ * from your code.
+ * To really start the replay you must call smpi_replay_run.
+ */
+void smpi_replay_init(int *argc, char ***argv)
+{
+  smpi_replay_initialize(argc, argv);
```

Comment:
Why do you need this function? Why don't you call smpi_replay_initialize() instead of smpi_replay_init() ?